examples/step-accident/lib/index.js

```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  ngModule: require('./angular'),
+  definition: {
+    code: 'accident-report-form',
+    name: "Accident Report Form",
+    description: "Form used to report vehicle accidents",
+    templates: {
+      form: "<accident-report-form></accident-report-form>",
+      view: "<accident-report value='result.submission'></accident-report>"
+    }
+  }
+};
```

Comment:
@wtrocki I pushed the step definitions into the steps as an export along with the angular module.

The bad thing is that it's hard for the portal app to import only the definition without including the entire step package (html templates, etc) even though the angular module itself won't be executed unless you call it.

A way to get around that would be to export the definition from a separate file (`require('step-module/definition.json')`), but that would require further changes in the file tree.

This might become a non-issue after the angular migration, since webpack should get rid of unused code once we use es6 exports.

Reply:
That's really heavy and it will push a lots of unused code. For the moment I think we will just duplicate them (copy step from step definion). I think that's easy to document and it's really a minor hassle. Angular migration is long stretch here - bringing entire module (with dependencies) for many steps will just extend build process and lead to 20mb of javascript bundles :D

Reply:
Another idea is to have interface instead of JSON here that will provide all requirements and applications will register steps in code rather than in the json file.

Reply:
I'm not sure what you mean by multiple schemas. The json schemas in the steps definitions would always be the same, even if for more configurable steps the schema is more complex (like including arrays and such).

Apart from i18n considerations I don't see why a step module can't provide it's own static definition. Not sure what you mean when you say code vs json file. code == runtime?

Reply:
Having a separate file with the definition would get rid of the extra code in the portal, since even with browserify everything else won't be included.

We only need to and an extra `index.js` to the root of the step modules and use it as the export point instead of using package.json#main to point deeper into the module, because it makes it weirder to make other files available via deeper requires.